### PR TITLE
Reuse the sqlalchemy engine once it's created

### PR DESCRIPTION
**Pull request by masenf**

### CODE REVIEW

**Review by Lendemor — DISMISSED** [review]

**Comment by masenf** [comment]
latest commit unifies the connection args and adds `pool_pre_ping` to avoid getting dead connections from the pool (now that we're not creating a new pool every time)

File: reflex/config.py [hunk 1]
```diff
@@ -568,6 +571,10 @@ class EnvironmentVariables:
 environment = EnvironmentVariables()
 
 
+# These vars are not logged because they may contain sensitive information.
+_sensitive_env_vars = {"DB_URL", "ASYNC_DB_URL"}
```
Comment (benedikt-bartscher) [inline reflex/config.py:575]: maybe we should add redis url here as well?

File: reflex/config.py [hunk 2]
```diff
@@ -748,7 +758,7 @@ def update_from_env(self) -> dict[str, Any]:
 
             # If the env var is set, override the config value.
             if env_var is not None:
-                if key.upper() != "DB_URL":
+                if key.upper() not in _sensitive_env_vars:
```
[the remark this answers is not in the document]
Comment (masenf) [reply reflex/config.py]: done

File: reflex/model.py [hunk 3]
```diff
@@ -425,6 +516,31 @@ def session(url: str | None = None) -> sqlmodel.Session:
...
+        url: The database url.
+
+    Returns:
+        An async database session.
+    """
+    global _AsyncSessionLocal
+    if url not in _AsyncSessionLocal:
+        _AsyncSessionLocal[url] = sqlalchemy.ext.asyncio.async_sessionmaker(
+            bind=get_async_engine(url),
+            class_=AsyncSession,
```
[the remark this answers is not in the document]
Comment (masenf) [reply reflex/model.py:537]: the default for `async_sessionmaker` is the `sqlalchemy.ext.asyncio.AsyncSession`, here we are using `sqlmodel.ext.asyncio.AsyncSession` which handles typing and scalar casting under the hood

**Pull request merged**